internal/js/modules/k6/browser/common/network_manager.go

```diff
@@ -881,10 +881,29 @@ func (m *NetworkManager) AbortRequest(requestID fetch.RequestID, errorReason str
 	return nil
 }
 
-func (m *NetworkManager) ContinueRequest(requestID fetch.RequestID) error {
+func (m *NetworkManager) ContinueRequest(
+	requestID fetch.RequestID,
+	opts ContinueOptions,
+	originalHeaders []HTTPHeader,
+) error {
 	m.logger.Debugf("NetworkManager:ContinueRequest", "continuing request (id: %s)", requestID)
-
 	action := fetch.ContinueRequest(requestID)
+
+	if len(opts.Headers) > 0 {
+		allHeaders := mergeHeaders(originalHeaders, opts.Headers)
+		action = action.WithHeaders(allHeaders)
+	}
+	if opts.URL != "" {
+		action = action.WithURL(opts.URL)
+	}
+	if opts.Method != "" {
+		action = action.WithMethod(opts.Method)
+	}
+	if opts.PostData != "" {
+		b64PostData := base64.StdEncoding.EncodeToString([]byte(opts.PostData))
+		action = action.WithPostData(b64PostData)
```

Comment:
I commented in the other place, but if this is the only place this is used, I guess we can directly base64 encode it and store it like that.

If that is a thing the user can access, IMO you might need more code around it being accessed correctly.

Reply:
Do you mean "encoding it in the `parseContinueOptions` function"? I don't think this is something the user should access (except if we implement the `fallback` function but this will be access through the `request` object so... code will have to change anyway).

Reply:
I guess - yes, but I do think it is probably fine not to do it as well. Just a random though I had.

 Arguably having the body as is and then encoding it as the protocol requires seems way more logical. Than haivng it in the exact format we will use it later, just because we do know this in advance.

Given that we have to encode this either way, I doubt there is any performance difference.